lib/actuators/src/actuator.h

```diff
@@ -0,0 +1,66 @@
+#include "filter.h"
+#include "drivers/actuator_drivers.h"
+#include "governors/governors.h"
+#include "converters/converters.h"
+
+class Actuator {
+  enum state_t{
+    ACTIVE,
+    ACTIVATING,
+    INACTIVE,
+    DEACTIVATING,
+    FAULT
+  };
+
+private:
+  state_t state;
+  int channel;
+  float target_value;
+  ActuatorDriver *driver;
+  Filter *filter;
+  Converter *converter;
+  Governor *governor;
+
+public:
+  //Constructor and destructor
+  Actuator(state_t s, int ch, ActuatorDriver *ad, Filter *f, Converter *c, Governor *g){
+    state = s;
+    channel = ch;
+    driver = ad;
+    filter = f;
+    converter = c;
+    governor = g;
+  }
+
+  ~Actuator(){};
+
+  //Getters and setters
+  state_t getState(){
+    return state;
+  }
+
+  void setState(state_t newState){
+    state = newState;
+  }
+
+  void set(float value){
+    if(governor->checkValue(value)){ //Validate value
+      target_value = value; //Update target value
+      float filtered_value = filter->filter(value); //Filter value
+      uint32_t converted_value = converter->convert(filtered_value); //Convert value from human readable
+
+      //If channel is 0, this is a DACtuator
```

Comment:
This logic is going to fail for channel 0 on the relays... unless we start offsetting everything by 1 🤕

Reply:
What about just setting channel to -1 for DACtuators then?

Reply:
alternatively, get rid of `set(x)` entirely and just have `set(x, ch)`  DACs have channels, just a single channel each (CH0).  The DACDriver can just ignore the provided channel number